src/Microsoft.Diagnostics.Monitoring.Options/S3StorageEgressProviderOptions.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.S3
+{
+    /// <summary>
+    /// Egress provider options for file system egress.
+    /// </summary>
+    internal sealed class S3StorageEgressProviderOptions :
```

Comment:
You'll want to add this to the [EgressOptions](https://github.com/dotnet/dotnet-monitor/blob/main/src/Microsoft.Diagnostics.Monitoring.Options/EgressOptions.cs) class so that the properties are available in the JSON schema.